openapi/nexus-internal.json

```diff
@@ -1546,6 +1546,11 @@
           }
         ]
       },
+      "ArtifactVersion": {
+        "description": "An artifact version.\n\nThis is a freeform identifier with some basic validation. It may be the serialized form of a semver version, or a custom identifier that uses the same character set as a semver, plus `_`.\n\nThe exact pattern accepted is `^[a-zA-Z0-9._+-]{1,63}$`.\n\n# Ord implementation\n\n`ArtifactVersion`s are not intended to be sorted, just compared for equality. `ArtifactVersion` implements `Ord` only for storage within sorted collections.",
+        "type": "string",
+        "pattern": "^[a-zA-Z0-9._+-]{1,63}$"
```

Comment:
Is the upper bound of 63 characters intended to prevent a hex-encoded 256-bit hash from being used, and if not, could we please increase it to 64? It seems to me that if we're going to allow freeform versions, we should accept arbitrary hashes of commonly used lengths, too.

Reply:
> Is the upper bound of 63 characters intended to prevent a hex-encoded 256-bit hash from being used, and if not, could we please increase it to 64? It seems to me that if we're going to allow freeform versions, we should accept arbitrary hashes of commonly used lengths, too.

Ah so the upper bound comes from database storage -- we decided in today's update watercooler to bump the limit to 64.

Reply:
Database schema updated in https://github.com/oxidecomputer/omicron/pull/7849.